src/main/java/use_case/autogen/AutogenInteractor.java

```diff
@@ -0,0 +1,135 @@
+            }
+            else{
+                presenter.prepareFailView("No valid timetable could be generated " +
+                        "with the given courses and constraints.");
+            }
+
+
+
+
+        } catch (Exception e) {
+            presenter.prepareFailView("An error occurred while generating the timetable: "
+                    + e.getMessage());
+        }
+
+    }
+
+    /**
+     *
+     * @return the search result containing the generated timetable (hopefully)
+     * If it's a fail it means no possible combinations found (atleast using DFS given current constraints)
+     */
+    public PotentialTimetable generateTimetable(List<CourseVariable> variables, List<Constraint> constraints) {
+        Set<Section> initial = new HashSet<>();
+        return timetableSearch(0, variables, constraints, initial);
+    }
+
+    /**
+     *
+     * @param index The current index number in the variables set that's having its section assigned
+     * @param assignment The assignment of sections (generated timetable) so far
+     * @param variables All course variables for the DFS (one per course)
+     * @param constraints The constraints that need to be satisfied
+     * @return A PotentialTimetable, if true contains a sucesfully generated timetable,
+     * else false, and it backtracks and tries a different path
+     */
+    private PotentialTimetable timetableSearch(int index, List<CourseVariable> variables, List<Constraint>
+            constraints, Set<Section> assignment) {
+        // This is the base case, that all variables have succesfully been assigned,
+        // and we have a fully generated timetable!
+        if(index == variables.size()){
+            return new PotentialTimetable(true, assignment);
+        }
+        //Choses the variable we will try finding a suitable lecture section for
+        CourseVariable var = variables.get(index);
+        for(Section candidate : var.getDomain()){ //Iterates through all possible candidates in variable's domain
+            //tries a possible assignment by adding a candidate to it
+            Set<Section> next = new HashSet<>(assignment);
+            next.add(candidate);
+
+            //checks if this new temporary assignment is valid
+            if (isConsistent(next, constraints)){
+                //if valid it checks if the rest of this path works (recursive step)
+                PotentialTimetable result = timetableSearch(index+1, variables, constraints, next);
+                //Valid path found then it succesfully returns this result so far
+                if (result.success){
+                    return result;
+                }
+            }
+        }
+        return new PotentialTimetable(false, Set.of()); //if no valid path found then returns false here so new-
+        //-path can be tried in the earlier recursive call
+
+    }
+
+
+    /** Checks if constraints are met, using the classes in the constraints package **/
+    /**
+     *
```

Comment:
invalid javadoc comment

```suggestion
    /**
     * Checks if constraints are met, using the classes in the constraints package
```